src/main/java/com/achdev/onlinebookstoreapp/exception/CustomGlobalExceptionHandler.java

```diff
@@ -26,30 +29,46 @@ protected ResponseEntity<Object> handleMethodArgumentNotValid(
             @NonNull HttpHeaders headers,
             @NonNull HttpStatusCode status,
             @NonNull WebRequest request) {
-        ValidationErrorResponseDto errorResponse = new ValidationErrorResponseDto();
-        errorResponse.setTimestamp(LocalDateTime.now());
-        errorResponse.setStatus(HttpStatus.BAD_REQUEST.getReasonPhrase());
+        HttpStatus errorStatus = HttpStatus.BAD_REQUEST;
+        ValidationErrorResponseDto exceptionResponseDto = createErrorResponse(
+                ValidationErrorResponseDto.class,
+                errorStatus.getReasonPhrase(),
+                "Validation failed for object");
         List<ErrorDetailDto> errorMessages = ex.getBindingResult().getAllErrors().stream()
                 .map(this::getErrorDetails)
                 .toList();
-        errorResponse.setErrors(errorMessages);
-        return new ResponseEntity<>(errorResponse, headers, HttpStatus.BAD_REQUEST);
+        exceptionResponseDto.setErrors(errorMessages);
+        return new ResponseEntity<>(exceptionResponseDto, headers, HttpStatus.BAD_REQUEST);
     }
 
     @ExceptionHandler(EntityNotFoundException.class)
-    protected ResponseEntity<Object> handleEntityNotFoundException(
-            EntityNotFoundException ex,
-            @NonNull WebRequest request) {
-        EntityNotFoundErrorResponseDto exceptionResponseDto =
-                new EntityNotFoundErrorResponseDto();
-        exceptionResponseDto.setTimestamp(LocalDateTime.now());
-        exceptionResponseDto.setStatus(HttpStatus.NOT_FOUND.getReasonPhrase());
-        exceptionResponseDto.setError("Entity not found");
-        exceptionResponseDto.setMessage(ex.getMessage() != null
-                ? ex.getMessage()
-                : "Object was not found");
-        exceptionResponseDto.setPath(request.getDescription(false));
-        return new ResponseEntity<>(exceptionResponseDto, HttpStatus.NOT_FOUND);
+    protected ResponseEntity<Object> handleEntityNotFoundException(EntityNotFoundException ex) {
```

Comment:
Suggestion (you can use such approach for other exceptions) and remove other error dto that you created
```suggestion
    @ExceptionHandler(EntityNotFoundException.class)
    protected ResponseEntity<Object> handleEntityNotFoundException(EntityNotFoundException ex) {
        return buildResponseEntity(new BookApiErrorResponse(NOT_FOUND,
                LocalDateTime.now(), ex.getMessage()));
    }
```

Reply:
ОК